### ISSUE TITLE
Survival/Drop Connect Rate (aka Stochastic Depth) is not implemented in tf.keras.applications.efficientnet

### ISSUE THREAD

**Issue by xhlulu** [tensorflow/tensorflow#45916]
In the original implementation, [the `drop_connect` function](https://github.com/tensorflow/tpu/blob/341c3b628409a03b74d2d66d33fd09532690e04e/models/official/efficientnet/utils.py#L276-L291) is implemented in the following way:

```python
def drop_connect(inputs, is_training, survival_prob):
  """Drop the entire conv with given survival probability."""
  # "Deep Networks with Stochastic Depth", https://arxiv.org/pdf/1603.09382.pdf
  if not is_training:
    return inputs

  # Compute tensor.
  batch_size = tf.shape(inputs)[0]
  random_tensor = survival_prob
  random_tensor += tf.random_uniform([batch_size, 1, 1, 1], dtype=inputs.dtype)
  binary_tensor = tf.floor(random_tensor)
  # Unlike conventional way that multiply survival_prob at test time, here we
  # divide survival_prob at training time, such that no addition compute is
  # needed at test time.
  output = tf.div(inputs, survival_prob) * binary_tensor
  return output
```

This function is used to decide whether to keep a block or not with a given probability `p_l`; this described in depth in [the original paper](https://arxiv.org/abs/1603.09382). 

Looking at the `tf.keras.applications.efficientnet` implementation, we can notice that the value is used as a dropout rate instead:

https://github.com/tensorflow/tensorflow/blob/5855c1c8ae2d2b36bf9d6f906b717046bb4afb36/tensorflow/python/keras/applications/efficientnet.py#L349-L353

https://github.com/tensorflow/tensorflow/blob/5855c1c8ae2d2b36bf9d6f906b717046bb4afb36/tensorflow/python/keras/applications/efficientnet.py#L413-L423

https://github.com/tensorflow/tensorflow/blob/5855c1c8ae2d2b36bf9d6f906b717046bb4afb36/tensorflow/python/keras/applications/efficientnet.py#L514-L517

The difference in implementation would be that the connection dropped is at a module level in the original implementation and at a unit level in the `tf.keras.applications` implementation.

**Comment by xhlulu** [comment]
I realized that the `noise_shape=(None, 1, 1, 1)` accomplishes the same as stochastic depth, so the keras implementation is indeed correct. My apologies.
